LDAR_Sim/model_code/ldar_sim_main.py

```diff
@@ -18,33 +18,60 @@
 # along with this program.  If not, see <https://opensource.org/licenses/MIT>.
 #
 # ------------------------------------------------------------------------------
-from pathlib import Path
 
-from batch_reporting import BatchReporting
-from ldar_sim_run import ldar_sim_run
-import pandas as pd
-import os
-import sys
-import shutil
-import datetime
-import warnings
 import multiprocessing as mp
-from generic_functions import check_ERA5_file
+import warnings
+import datetime
+import shutil
+import os
+import pandas as pd
+import argparse
+from argparse import RawTextHelpFormatter
 
+from ldar_sim_run import ldar_sim_run
+from batch_reporting import BatchReporting
+from pathlib import Path
 from input_manager import InputManager
+from generic_functions import check_ERA5_file
 
 if __name__ == '__main__':
+    # Get route directory , which is parent folder of ldar_sim_main file
+    # Set current working directory directory to root directory
+    root_dir = Path(os.path.dirname(os.path.realpath(__file__))).parent
     # ------------------------------------------------------------------------------
     # Look for parameter files supplied as arguments - if parameter files are supplied as
     # arguments, proceed to parse and type check input parameter type with the input manager
-    parameter_filenames = sys.argv[1:]
+    # will also accept flagged input directory , (-P or --in_dir)
+    parser = argparse.ArgumentParser(formatter_class=RawTextHelpFormatter)
+    # ---Declare input arguments---
+    parser.add_argument(
+        'in_files', type=str, nargs='*',
+        help='Input files, seperate with space, can be absolute path or relative to' +
+        'root directory (LDAR_Sim). All files should have yaml, yml, or json extensions \n' +
+        'ie. python ldar_sim_main.py ./file1.json c:/path/to/file/file2.json')
+    parser.add_argument(
+        "-P", "--in_dir",
+        help='Input Directory, folder containing input files, will input all files within' +
+        'folder that have yaml, yml or json extensions \n' +
+        'ie. python ldar_sim_main.py --in_dir ./folder_with_infiles')
+    args = parser.parse_args()
+
+    if args.in_dir is not None:
+        # if an input directory is specified, get all files within that are in the directory
+        in_dir = root_dir / args.in_dir
+        # Get all yaml or json files in specified folder
+        parameter_filenames = [
+            "{}/{}".format(args.in_dir, f) for f in os.listdir(in_dir)
+            if ".yaml" in f or ".yml" in f or ".json" in f]
+    else:
+        parameter_filenames = args.in_files
```

Comment:
How to handle the ordering of parameter files here? The order matters as each set updates what exists previously. Ordering is opaque and I think the os module just asks the filesystem, which makes this OS dependent as linux and windows index files differently.

Should we:
- expand documentation
- just roll forward as is and don't worry about it (not a problem if there is no parameter file stacking on specific programs or modules)
- add a `sorted` or something to allow sorting to be done via normal alphanumeric sorting in a directory or prefixes: e.g., `1_globals.yaml`, etc.
- or . remove this functionality because it is has potential for poorly defined behaviour?

Reply:
@tbarchyn  Ahhh! Yeah good point on the order. I really just added this to speed up my testing and it is totally lazy, or atleast bringing in all parameters. This was kind of brought up in last weeks meeting about how to define all the files a bit easier, especially when running in debug mode.

Reply:
@tbarchyn how do you feel about declairing the files to import directly in the global parameter file:

ie: 
(Global yaml)
parameter_level: global
version: '2.0'
programs:
- P_air.yaml # maybe do not even include extension and get program to handle extension.
- P_ref.yaml 
- P_base.yaml 

Other option is using a value either in the file name or within the file itself?